components/set-password.tsx

```diff
@@ -0,0 +1,139 @@
+"use client";
+
+import React from "react";
+import AuthenticationPageTemplate from "@/components/authentication-page-template";
+import { verifiedSignUpFormDetails } from "@/utils/forms/initial-values";
+import { VerifiedSignUpFormDetails } from "@/utils/forms/interfaces";
+import { verifiedSignUpValidationSchema } from "@/utils/forms/validations";
+import { Button } from "@/components/ui/button";
+import { verifiedSignUpForm } from "@/utils/forms/form-details";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormMessage,
+} from "@/components/ui/form";
+import { useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { useSetPasswordMutation } from "@/redux/apis/auth-api";
+import { useRouter, useSearchParams } from "next/navigation";
+import { ROUTES } from "@/utils/constants";
+import { useDispatch } from "react-redux";
+import { toast } from "sonner";
+import { loginUser } from "@/redux/slice/user-slice";
+
+const SetPassword = ({ type }: { type: "INVITE" | "RESET" }) => {
+  const [setupPassword, { isLoading }] = useSetPasswordMutation();
+
+  const searchParams = useSearchParams();
+  const token = searchParams.get("token");
+
+  const router = useRouter();
+  const dispatch = useDispatch();
+
```

Comment:
use this order instead

1- Define Hooks
2- Define States
3- Define getters i.e searchParam or localStorage , Redux etc.
4- Define useEffects


use this instead 

```
const router = useRouter();
const dispatch = useDispatch();  
const searchParams = useSearchParams();
const [setupPassword, { isLoading }] = useSetPasswordMutation();
const token = searchParams.get("token");
```